File: sample-rest-api/src/main/java/nl/haploid/octowight/sample/controller/CaptainController.java
package nl.haploid.octowight.sample.controller;

import nl.haploid.octowight.sample.data.CaptainModel;
import nl.haploid.octowight.sample.service.CaptainService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import java.util.List;

@Controller
@RequestMapping("/captain")
@ResponseBody
public class CaptainController extends AbstractController {

	@Autowired
	private CaptainService captainService;

	@RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
	public List<CaptainModel> getCaptains() {
		return captainService.getAllModels();
	}

	@RequestMapping(value = "/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
	public CaptainModel getCaptain(final @PathVariable long id) {
		return captainService.getModel(id);
	}
}

File: sample-rest-api/src/main/scala/nl/haploid/octowight/sample/controller/CaptainController.scala
package nl.haploid.octowight.sample.controller

import nl.haploid.octowight.sample.service.CaptainService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.{PathVariable, RequestMapping, RequestMethod, ResponseBody}

import scala.collection.JavaConverters._

@Controller
@RequestMapping(Array("/captain"))
@ResponseBody
class CaptainController extends AbstractController {
  @Autowired private[this] val captainService: CaptainService = null

  @RequestMapping(method = Array(RequestMethod.GET), produces = Array(MediaType.APPLICATION_JSON_VALUE))
  def getCaptains = captainService.getAllModels.toList.asJava

  @RequestMapping(value = Array("/{id}"), method = Array(RequestMethod.GET), produces = Array(MediaType.APPLICATION_JSON_VALUE))
  def getCaptain(@PathVariable id: Long) = captainService.getModel(id)
}
